build/Publish.targets

```diff
@@ -200,7 +201,8 @@
                     AccountKey="$(PublishBlobFeedKey)"
                     ItemsToPush="@(PackagesToPublishToTransport)"
                     Overwrite="$(PushToBlobFeed_Overwrite)"
-                    ManifestBuildId="$(BuildNumber)"
+                    ManifestBuildId="$(Version)"
+                    ManifestBuildData="ProductVersion=$(PackageVersion);UniverseCommitHash=$(CommitHash)"
```

Comment:
Out of curiosity, what's `Version` vs. `PackageVersion` vs. `BuildNumber`?

If `UniverseCommitHash` is kinda to open the way to have multiple hashes listed in one Build element, originally my thinking was that each "root" repo (which may contain submodules) would be one build. But I know now there are multiple "roots" that go into one aspnetci build, so this makes sense. Unfortunately the build metadata (`ManifestBuildId`, `ManifestBuildData`, ...) only gets set on the first update, but that could be fixed to start merging it if you're looking to add more Build metadata over the course of the build.

Reply:
So, for our final builds, the values might look like this:
```
BuildNumber    = 1234
Version        = 2.1.0-preview1-1234
PackageVersion = 2.1.0-preview1-final
```

Re: UniverseCommitHash

I wasn't looking to add more at the moment. I was just looking to make it explicit in the manifest what "Commit" means.

Reply:
Just to make sure, the `2.1.0-preview1-final` matches what you'll be publishing? That is, each final build will need to overwrite every aspnet asset on `dotnetcli`?

Reply:
Yes, 2.1.0-preview1-final is the "product friendly version", the version of nupkgs, installers, etc. `Version` is the more like a build ID. It represents additional info that is sometimes baked in to assembly attributes and other things.